doc(custom element): add custom element basics doc

AshleyGrant opened the pull request:
Let's consider this a first draft of this documentation. 

Questions:
Are there any other topics that need to be covered in the "basics" documentation?
Should we add working examples to go along with the example code?
Is it okay that I'm planning on splitting custom element documentation out in to multiple files?
Why is "A Christmas Story" the greatest Christmas movie of all time, and how did I watch Top Gun while writing this, but come up with examples that didn't involve Maverick?

EisenbergEffect commented:
I think I'd rather have it all in one doc. We have a TOC UI that is built based on the section titles. We are improving that int he next version of the docs app to make it more visible.

EisenbergEffect commented:
This is a good start. A couple of notes before moving on with additional documentation:

* There's a typo where you say `customAttribute` on accident.
* I'd prefer to not use the Skeleton application when explaining things. We may change that in the future or it may go away entirely and I'd rather the docs not be dependent on that as an example.

Other than that, it looks pretty good. If you can address those two issues, we can merge this in and then you can pick up with adding more details.

AshleyGrant commented:
Alright, I fixed the typos where I used `customAttribute` and changed the wording to not mention the Skeleton project. I also changed the file name to not say basics.
